### PR TITLE
Application connectivity validator performance improvements

### PR DESCRIPTION
**Pull request by koala7659**
**Performance and scalability improvements:**

- Handler fix - stop building `Regexp` object every time when request is processed
- Use custom `http.Transport` to build reverse proxy instead of the default one - allows for tcp connections reuse
- Fill cache with application data from all the applications before servers starts - should eliminate 404 cache miss errors after scaling the service 
- Extend cached data for application by adding target eventing endpoints. Do not create them on the fly with each request.

- Unit test adjustments 

**Results**: 
Smaller latency, much less resource consumption, much better throughput

200 seconds performance tests on single pod Kyma Azure Dev :

```
40000 events  (200 events/sec)             

                     Old version                         New version:
---------------------------------------------------------------------------------------------
validator-pod        CPU: 221m, Memory: 19Mi      CPU: 153m, Memory: 18Mi
istio-proxy          CPU: 199m, Memory: 54Mi      CPU: 101m, Memory: 58Mi
Avg Latency:         0.0178s                      0.0146s
Errors:              0                            0
---------------------------------------------------------------------------------------------


 
60000 events (300 events/sec)  
 
                     Old version                         New version:
---------------------------------------------------------------------------------------------
validator-pod        CPU: 493m, Memory:  221m      CPU: 224m, Memory: 18Mi
istio-proxy          CPU: 199m, Memory: 56Mi       CPU: 163m, Memory: 51Mi                                
Avg Latency:         0.0351s                       0.0195s
Errors:              0                             0
---------------------------------------------------------------------------------------------

 
100000 events (500 events/sec )  

                     Old version                         New version:
---------------------------------------------------------------------------------------------
validator-pod        CPU: 1008m, Memory: 784Mi         CPU: 350m, Memory: 21Mi
istio-proxy          CPU: 373m,  Memory: 805Mi         CPU: 247m, Memory: 53Mi
Avg Latency:         0.6604s                           0.0310s
Errors:              75250 (76%)                       0        
---------------------------------------------------------------------------------------------


150000 events (750 events/sec )  

                     Old version                         New version:
---------------------------------------------------------------------------------------------
validator-pod        X (pod crashed)                   CPU: 477m, Memory: 43Mi
istio-proxy          X (pod crashed)                   CPU: 364m, Memory: 63Mi
Avg Latency:         0.3465s                           0.0460s
Errors:             101247 (68%)                       0        
---------------------------------------------------------------------------------------------

200000 events (1000 events/sec )  

                     Old version                         New version:
---------------------------------------------------------------------------------------------
validator-pod        X (pod crashed)                   CPU: 604m, Memory: 77Mi
istio-proxy          X (pod crashed)                   CPU: 467m, Memory: 70Mi
Avg Latency:         1.0383s                           0.0974s
Errors:              172715 (87%)                      0        
---------------------------------------------------------------------------------------------

```

### CODE REVIEW

File: components/central-application-connectivity-validator/internal/controller/cachesynccontroller.go [hunk 1]
```diff
@@ -31,6 +31,9 @@ func (c *controller) Reconcile(ctx context.Context, request reconcile.Request) (
 }
 
 func (c *controller) SetupWithManager(mgr ctrl.Manager) error {
+
+	c.cacheSync.Init(context.Background())
```
[the remark this answers is not in the document]
Comment (koala7659) [reply components/central-application-connectivity-validator/internal/controller/cachesynccontroller.go:35]: Finally error will not be returned . I will start reconciler no matter if Init functions succeeds

**Review by m00g3n — APPROVED** [review]
LGTM

**Pull request merged**